cmd/tidb-dashboard/main.go

```diff
@@ -63,36 +80,58 @@ type DashboardCLIConfig struct {
 	KVFileEndTime   int64
 }
 
-// NewCLIConfig generates the configuration of the dashboard in standalone mode.
-func NewCLIConfig() *DashboardCLIConfig {
-	cfg := &DashboardCLIConfig{}
-	cfg.CoreConfig = &config.Config{}
+func getContext() context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		sc := make(chan os.Signal, 1)
+		signal.Notify(sc,
+			syscall.SIGHUP,
+			syscall.SIGINT,
+			syscall.SIGTERM,
+			syscall.SIGQUIT)
+		<-sc
+		cancel()
+	}()
+	return ctx
+}
 
-	var showVersion bool
-	flag.BoolVar(&showVersion, "v", false, "Print version information and exit")
-	flag.BoolVar(&showVersion, "version", false, "Print version information and exit")
-	flag.StringVar(&cfg.ListenHost, "host", "0.0.0.0", "The listen address of the Dashboard Server")
-	flag.IntVar(&cfg.ListenPort, "port", 12333, "The listen port of the Dashboard Server")
-	flag.StringVar(&cfg.CoreConfig.DataDir, "data-dir", "/tmp/dashboard-data", "Path to the Dashboard Server data directory")
-	flag.StringVar(&cfg.CoreConfig.PathPrefix, "path-prefix", "/dashboard", "Dashboard URL prefix")
-	flag.StringVar(&cfg.CoreConfig.PDEndPoint, "pd", "http://127.0.0.1:2379", "The PD endpoint that Dashboard Server connects to")
-	flag.BoolVar(&cfg.EnableDebugLog, "debug", false, "Enable debug logs")
-	// debug for keyvisual，hide help information
-	flag.Int64Var(&cfg.KVFileStartTime, "keyviz-file-start", 0, "(debug) start time for file range in file mode")
-	flag.Int64Var(&cfg.KVFileEndTime, "keyviz-file-end", 0, "(debug) end time for file range in file mode")
+func buildTLSConfig(caPath, keyPath, certPath *string) *tls.Config {
+	tlsInfo := transport.TLSInfo{
+		TrustedCAFile: *caPath,
+		KeyFile:       *keyPath,
+		CertFile:      *certPath,
+	}
+	tlsConfig, err := tlsInfo.ClientConfig()
+	if err != nil {
+		log.Fatal("Failed to load certificates", zap.Error(err))
+	}
+	return tlsConfig
+}
 
-	clusterCaPath := flag.String("cluster-ca", "", "path of file that contains list of trusted SSL CAs.")
-	clusterCertPath := flag.String("cluster-cert", "", "path of file that contains X509 certificate in PEM format.")
-	clusterKeyPath := flag.String("cluster-key", "", "path of file that contains X509 key in PEM format.")
+func init() {
+	cfg.CoreConfig = &config.Config{}
+	rootCmd.Version = utils.ReleaseVersion
+
+	rootCmd.Flags().StringVar(&cfg.ListenHost, "host", "0.0.0.0", "The listen address of the Dashboard Server")
```

Comment:
I think you'd better arrange `run` related parameters in `runCmd` and `kvauth` related parameters in `kvauthCmd`, instead of here, since they are not truly "global".

Reply:
got it.　